src/frontend/src/utils/work-package.utils.ts

```diff
@@ -11,3 +11,23 @@ export const getTitleFromFormType = (formType: WPFormType, wbsElement: WbsElemen
       return `${wbsPipe(wbsElement.wbsNum)} - ${wbsElement.name}`;
   }
 };
+
+/**
+ * Given a list of work packages, return the work packages that are overdue.
+ * @param wpList a list of work packages.
+ * @returns a list of work packages that are overdue.
+ */
+export const getOverdueWorkPackages = (wpList: WorkPackage[]): WorkPackage[] => {
+  const overdueWorkPackages: WorkPackage[] = [];
+
+  for (let i = 0; i < wpList.length; i++) {
+    const curr = wpList[i];
+
+    // if the work package is anything but complete and the end date is before today, it is overdue.
+    if (curr.status !== WbsElementStatus.Complete && curr.endDate < new Date()) {
+      overdueWorkPackages.push(curr);
+    }
+  }
+
```

Comment:
This works perfectly! I think a cleaner way to do this would be to do something like filter the list, it'll make the code a little easier to read